### ISSUE TITLE
« Reconcile > Use values as identifiers » can match an invalid ID and will show JSON data instead of a nice error message

### ISSUE THREAD

**Issue by antoine2711** [OpenRefine/OpenRefine#4715]
The function « Reconcile > Use values as identifiers » can match an invalid ID and will show JSON data that is of little interest and can confuse a non-technical user.

A nice error message would be much better.

Note that this is only for Wikidata, as @WaltonG in #PR4666 found, these results and the others already have a nice error message:
> Wikidata - {"arguments":{"id":"93","lang":"en"},"details":"'93'","message":"invalid query","status":"error"}
> GND - Not found : 93
> VIAF- No preview
> OpenData - No preview
> ORCID - No preview

### To Reproduce
1. « Reconcile > Use values as identifiers » on an invalid wikidata ID
2. Move the mouse cursor over the reconciled ID
3. Finally, … well, decode visually the JSON and try to make some sense out of it…

### Current Results
A JSON string.

### Expected Behavior
A nice error message.
Wikidata identifiant with letters or… should be refused at the « Reconcile > Use values as identifiers » stage.

### Screenshots
![image](https://user-images.githubusercontent.com/12217525/162355341-39dbea64-56b5-4491-b22f-a8ae19c3416e.png)

![image](https://user-images.githubusercontent.com/12217525/162356984-18aa5018-2a33-4382-a9d5-d1f1c9eabd57.png)

### Versions
 - Operating System: All
 - OpenRefine: v3.5.x

**Comment by dikshamalik2552** [comment]
Hi @antoine2711, Can you please assign me this issue

**Comment by wetneb** [comment]
I think this is an issue with the reconciliation service, not with OpenRefine, so I would close this. But happy to leave open if you can think of a sensible fix on OpenRefine's side.

**Comment by antoine2711** [comment]
@wetneb: is there a way that « Use values as identifiers » can validate the ID structure? So that OR can refuse to match invalid ID like W12wretret2 for WD recon service? Like getting a grep from the service?

Regards, Antoine.

**Comment by wetneb** [comment]
In my opinion, the whole point of this operation is that it does not do HTTP calls to the service: it is meant to be fast. If one wants to validate the reconciliation identifiers, one should use the Reconcile operation instead.

**Comment by antoine2711** [comment]
> In my opinion, the whole point of this operation is that it does not do HTTP calls to the service: it is meant to be fast.

Yes @wetneb, that's why a REGEX validation would be good in my opinion: fast, and still give better error management. I was just asking if the REGEX validation for Wikidata ID was available from the Recon Service.

If not, could we assume that Wikibase ID are always ONE LETTER + NUMBERS?

Regards, Antoine

**Comment by antoine2711** [comment]
Another example of an error preview, from a good item:
![image](https://user-images.githubusercontent.com/12217525/162836992-30b105aa-916d-48a3-b807-67807761cb79.png)

Regards, Antoine

**Comment by wetneb** [comment]
Removing the "good first issue" label because there is no clear and consensual plan about how to fix this.

**Comment by tfmorris** [comment]
> In my opinion, the whole point of this operation is that it does not do HTTP calls to the service: it is meant to be fast. If one wants to validate the reconciliation identifiers, one should use the Reconcile operation instead.

This sounds like a premature optimization to me. A fast, but incorrect, service if of limited usefulness. Validating a native ID should be a blindingly fast operation. The Freebase equivalent to this did HTTP calls and was responsive.

Having said that, nicer error formatting on preview errors would definitely be useful.

**Comment by thadguidry** [comment]
We still have both 2 things, correct?  Where 1 depends on the recon service utilized, but still...

1. A way to make an HTTP call to validate an ID exists.
In Wikidata, that endpoint would be like this example `https://wikibase.example/w/rest.php/wikibase/v0/entities/items/Q178794` and which returns when not found
 ```
 {
  "code": "invalid-item-id",
  "message": "Not a valid item ID: {item_id}"
}
```
2. A way to skip making any HTTP calls and assume that all ID's for recon purposes are already valid.

**Comment by tfmorris** [comment]
Validating a single ID per HTTP call *will* be slow, because you'll be drowning in latency. The service should allow for batches of IDs to be verified like Freebase did. Wikidata may have such a call, but the one above (https://wikibase.example/w/rest.php/wikibase/v0/entities/items/Q178794) isn't it.
